nd4j/samediff-import/samediff-import-onnx/src/main/kotlin/org/nd4j/samediff/frameworkimport/onnx/definitions/implementations/Expand.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ *  ******************************************************************************
+ *  *
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  *  See the NOTICE file distributed with this work for additional
+ *  *  information regarding copyright ownership.
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+package org.nd4j.samediff.frameworkimport.onnx.definitions.implementations
+
+import org.nd4j.autodiff.samediff.SDVariable
+import org.nd4j.autodiff.samediff.SameDiff
+import org.nd4j.autodiff.samediff.internal.SameDiffOp
+import org.nd4j.ir.OpNamespace
+import org.nd4j.linalg.api.buffer.DataType
+import org.nd4j.linalg.factory.Nd4j
+import org.nd4j.samediff.frameworkimport.hooks.PreImportHook
+import org.nd4j.samediff.frameworkimport.hooks.annotations.HookResult
+import org.nd4j.samediff.frameworkimport.hooks.annotations.PreHookRule
+
+/**
+ * A port of expand.py from onnx tensorflow for samediff:
+ * https://github.com/onnx/onnx-tensorflow/blob/master/onnx_tf/handlers/backend/expand.py
+ *
+ * @author Adam Gibson
+ */
+@PreHookRule(nodeNames = [],opNames = ["Expand"],frameworkName = "onnx")
+class Expand : PreImportHook  {
+    override fun preProcess(
+        op: SameDiffOp,
+        sd: SameDiff,
+        attributes: Map<String, Any>,
+        descriptor: OpNamespace.OpDescriptor,
+        outputNames: List<String>,
+        isFinalOutput: Boolean
+    ): HookResult {
+        // Parameter docs below are from the onnx operator docs:
```

Comment:
@treo as documented this is a port of tf onnx's mapping which follows pretty closely with what we did for TF.